src/controller/order-controller.ts

```diff
@@ -127,3 +162,66 @@ export async function captureOrderController(fastify: FastifyInstance) {
     },
   });
 }
+
+// Patch order
+export type ShippingOption = {
+  id: string;
+  label: string;
+  type: string;
+  selected: boolean;
+  amount: {
+    value: string;
+    currency_code: string;
+  };
+};
+
+async function patchOrderHandler(request: FastifyRequest, reply: FastifyReply) {
+  const { orderID } = request.body as {orderID: string};
+  const { access_token: accessToken } = await getAuthToken();
+
+  const data = await onShippingChange(accessToken, orderID);
+  if (JSON.stringify(data) !== '{}') {
+    reply.send(data);
+  } else {
+    // Send a response with a 204 status code and no body
+    reply.code(204).send();
+  }
+}
+
+export async function patchOrderController(fastify: FastifyInstance) {
+  fastify.route({
+    method: "PATCH",
+    url: "/patch-order",
+    handler: patchOrderHandler,
+    schema: {
+      body: {
+        type: "object",
+        required: ["orderID"],
+        properties: {
+          orderId: {
+            type: "string",
+          },
+        },
+      },
+    },
+  });
+}
+
+//retrieve order details
+async function getOrderHandler(
+  request: FastifyRequest,
+  reply: FastifyReply
+) {
+  const { access_token: accessToken } = await getAuthToken();
+  const { orderID } = request.body as { orderID: string };
+  const data = await getOrder(accessToken, orderID);
+  reply.send(data);
+}
+
+export async function getOrderController(fastify: FastifyInstance) {
+  fastify.route({
+    method: "GET",
```

Comment:
This might need to be a POST, if we're expecting a request body in the handler.

Reply:
I think a GET is fine here to match the backend orders api format.